signer/src/testing/btc.rs

```diff
@@ -89,3 +92,13 @@ pub async fn new_zmq_block_hash_stream(endpoint: &str) -> ReceiverStream<Result<
 
     ReceiverStream::new(receiver)
 }
+
+/// Return the canonical (tallest) chain tip from `get_chain_tips`
+pub fn get_canonical_chain_tip(rpc: &Client) -> GetChainTipsResultTip {
+    rpc.get_chain_tips()
+        .unwrap()
+        .iter()
+        .max_by_key(|t| t.height)
+        .unwrap()
+        .clone()
+}
```

Comment:
Now we can have multiple tips on bitcoin, so popping from `get_chain_tips` was returning the wrong one. Looking at the docs it doesn't seem `getchaintips` rpc guarantee a specific ordering on the results. We could switch this in favor of searching for the `active` tip instead of the tallest.

Reply:
Yeah, we should go with `active` tip, since that is the canonical one.

Reply:
Done in [a101cd8](https://github.com/stacks-sbtc/sbtc/pull/1585/commits/a101cd86bcf0c094b231f376fb3d331658edd644)